pkg/virt-controller/watch/drain/disruptionbudget/disruptionbudget.go

```diff
@@ -340,7 +344,16 @@ func (c *DisruptionBudgetController) sync(key string, vmi *virtv1.VirtualMachine
 			delete = true
 		} else if wantsToMigrate && vmi.DeletionTimestamp == nil && pdb == nil {
 			// No pdb and we want migrations on evictions
+			minAvailable = intstr.FromInt(1)
 			create = true
+		} else if wantsToMigrate && migrationInProgress(vmi) {
```

Comment:
Is there a gap here we have to worry about, where the source VMI could get evicted if the `minAvailable=1` and the target pod comes online for the migration?

so

- eviction is triggered
- minAvailable = 1 prevents VMI pod from being removed
- migration is started
- targetPod comes online 

At this point there are 2 VMI pods and minAvailable = 1. It isn't until after the migration starts that we set minAvailable=2 which seems like there might be a period of time where it's possible for the source VMI pod to get evicted.

Reply:
In theory yes. We should cover it. I would suggest to also watch the PDBs for the VMs in the migration controller and only start the migration if we see a `2` on the PDB for a VMI. Sounds like the easiest solution.

Reply:
ha, chicken and egg problem as the structure exists today.  pbd is updated to 2 when migration is in progress, migration can only be started when pbd is 2.

Reply:
Two other options which come immediately to my mind:

 1) Merging the controllers (now they seem to belong much more together than before)
 2) Finding out why `spec.maxUnavailable` can't be used. Maybe just the documentation did not get updated.

Reply:
I think that we could also make the migration controller ware of the PDB and wait with the pod rendering until the PDB has been extended. It seems like the quickest solution at this point.